src/middlewared/middlewared/etc_files/smartd.py

```diff
@@ -71,6 +71,8 @@ def get_smartd_schedule(disk):
 
 
 def get_smartd_schedule_piece(value, min, max):
+    if value == "*":
+        return "."
```

Comment:
I think this won't work when being used for e.g. `daymonth` because it'll require `..` instead of `.` for 10, 11, 12, ...

I `return "." * len(str(max))` for cases like this.

You can also add corresponding unit test.